doc/query.md

```diff
@@ -39,27 +39,60 @@ Secondly, these documents are in the datastore:
+  "comment": "Mike goes to middle school and likes reading books." };
 
 { "name": "mike", 
   "age": 34, 
-  "pet": {"species": "dog"} };
+  "pet": {"species": "dog"},
+  "comment": "Mike is a doctor and likes reading books." };
 
 { "name": "fred", 
   "age": 23, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Fred works for a startup out of his home office." };
 ```
 
-### Creating indexes
+### Creating Indexes
 
 In order to query documents, indexes need to be created over
-the fields to be queried against.
+the fields to be queried against.  Currently we support two types of indexes.  The first, a JSON index, is used by query clauses containing comparison operators like `$eq`, `$lt`, `$gt`, etc.  Query clauses containing these operators rely on standard SQL syntax along with JSON indexes to provide query results.  The second, a TEXT index, uses SQLite's full text search (FTS) engine.  A query clause containing a `$text` operator with a `$search` operator uses [SQLite FTS SQL syntax] [ftsHome] along with a TEXT index to provide query results.
 
-Use `ensureIndexed(List<Object> fieldNames, String indexName)` to create indexes. These indexes are persistent across application restarts as they are saved to disk. They are kept up to date documents change; there's no need to call `ensureIndexed(List<Object> fieldNames, String indexName)` each time your applications starts, though there is no harm in doing so.
+Using the datastore documents example above, a query like `{ "name" : { "$eq" : "mike" } }` would benefit from a JSON index on the `name` field.  Whereas a query like `{ "$text" : { "$search" : "doctor books" } }` would _require_ a TEXT index on the `comment` field but could include as many fields as desired.  In other words, A TEXT index can be used to search for terms, phrases, prefixes with wildcards, etc.  Query syntax details are covered later in this document.
 
-The first argument to `ensureIndexed(List<Object> fieldNames, String indexName)` is a list of fields to put into this index. The second argument is a name for the index. This is used to delete indexes at a later stage and appears when you list the indexes in the database.
+[ftsHome]: http://www.sqlite.org/fts3.html#section_3
 
-A field can appear in more than one index. The query engine will select an appropriate index to use for a given query. However, the more indexes you have, the more disk space they will use and the greater overhead in keeping them up to date.
+
+
+Use the following to create a JSON index.
+
+```
+ensureIndexed(List<Object> fieldNames, 
+              String indexName)
+```
+
+Use either one of the following to create a TEXT index.
+
+```
+ensureIndexed(List<Object> fieldNames, 
+              String indexName, 
+              String indexType)
+``` 
+
+or 
+
+```
+ensureIndexed(List<Object> fieldNames, 
+              String indexName, 
+              String indexType, 
+              Map<String, String> indexSettings)
+``` 
```

Comment:
Colon after "create a TEXT index" rather than a full stop.

I would put them both in one code block with the `or` commented in the middle.